Kernel/FileSystem/MountFile.cpp

```diff
@@ -0,0 +1,116 @@
+/*
+ * Copyright (c) 2022, Liav A. <liavalb@hotmail.co.il>
+ *
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+
+#include <AK/StringView.h>
+#include <Kernel/API/FileSystem/MountSpecificFlags.h>
+#include <Kernel/API/Ioctl.h>
+#include <Kernel/API/POSIX/errno.h>
+#include <Kernel/API/POSIX/unistd.h>
+#include <Kernel/FileSystem/Inode.h>
+#include <Kernel/FileSystem/MountFile.h>
+#include <Kernel/FileSystem/OpenFileDescription.h>
+#include <Kernel/FileSystem/VirtualFileSystem.h>
+#include <Kernel/Memory/PrivateInodeVMObject.h>
+#include <Kernel/Memory/SharedInodeVMObject.h>
+#include <Kernel/Process.h>
+#include <Kernel/StdLib.h>
+
+namespace Kernel {
+
+ErrorOr<NonnullLockRefPtr<MountFile>> MountFile::create(FileSystemInitializer const& file_system_initializer, int flags)
+{
+    // NOTE: We should not open a MountFile if someone wants to either remount or bindmount.
+    // There's a check for this in the fsopen syscall entry handler, but here we just assert
+    // to ensure this never happens.
+    VERIFY(!(flags & MS_BIND));
+    VERIFY(!(flags & MS_REMOUNT));
+    auto mount_specific_data_buffer = TRY(KBuffer::try_create_with_size("Mount Specific Data"sv, 4096, Memory::Region::Access::ReadWrite, AllocationStrategy::AllocateNow));
```

Comment:
What's the magic value `4096`? Is it `PAGESIZE`?

Reply:
Yes it is, I should use that.